### PR TITLE
[SP-4603] Remove unused code and not needed null checks

### PR DESCRIPTION
**Pull request by Suryakantj**

### CODE REVIEW

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java [hunk 1]
```diff
@@ -415,14 +408,12 @@ void showView(View view, boolean isFromPM) {
     }
 
     private void runPMReady(){
-        if (this.pmReady != null)
-            uiThreadHandler.postIfEnabled(GDPRConsentLib.this.pmReady::run);
+        uiThreadHandler.postIfEnabled(GDPRConsentLib.this.pmReady::run);
```
Comment (leodoin) [inline cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:411]: @Suryakantj , have you tested this without setting the `onPmReady` callback ?

**Review by leodoin — APPROVED** [review]
To be on the safe side it's better to double check (right some tests) to ensure we are not throwing erros on the optional callback are not set by the user.

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java (hunk 1 above)
Comment (Suryakantj) [reply cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:411]: @leodoin I did test without setting those callbacks and it works as expected.

**Comment by Suryakantj** [comment]
> To be on the safe side it's better to double check (right some tests) to ensure we are not throwing erros on the optional callback are not set by the user.

I have added unit tests for optional callback

**Review by leodoin — APPROVED** [review]
👍

**Pull request merged**